source/developers/how-tos/enable-translations-new-repo.rst

```diff
@@ -5,21 +5,196 @@ Enabling Translations on a New Repo
+
+To include a repo in the translations workflow follow the steps below:
+
+1. If you are not working within an existing repository, create a new repo using the: `edx-cookiecutters`_
+   python-library for either ``django-app`` or ``xblock``.
+   If the repo already exists continue to the next steps.
+
+2. For existing repos which don't have the ``make extract_translations`` command, it can be copied from the
+   `edx-cookiecutters`_ Makefile in the corresponding template e.g. ``cookiecutter-xblock`` Makefile for XBlocks.
+
+   **Note:** Some repositories uses ``django-manage makemessages``. The recommendation is to use ``i18n_tool extract``
+   because it provides more options and cleans ``.po`` files.
+   However, it should be fine to use either as long as the translation files are extracted and working .
+
+3. Run the ``make extract_translations``. Verify that it extracts the files within the ``my_xblock_module/conf/locale``
+   directory. The PO file names will vary depending on the use case:
+
+   - The XBlocks will have a ``text.po`` extracted.
+   - Django plugins will have ``django.po``.
+   - Other plugins may include ``djangojs.po`` if the repo has a ``static`` directory with JavaScript use of
+     ``gettext``.
+
+4. Add the repository to `extract-translation-source-files.yml`_ in the `openedx-translations repo`_.
+
+   Add a new entry under the ``python-translations`` section. For example for the XBlock located in
+   ``https://github.com/openedx/MyXBlock`` and has the XBlock Python module name ``my_ai_bot_xblock`` it should have
+   the following entry::
+
+    python-translations:
+      strategy:
+        matrix:
+          repo:
+            ...
+            - repo_name: MyAIBotXBlock
+              python_module_name: my_ai_bot_xblock
+            ...
+
+   When adding a theme or any other module that's not an Open edX plugin or an XBlock the entry should
+   added without ``python_module_name`` as demonstrated below::
+
+    python-translations:
+      strategy:
+        matrix:
+          repo:
+            ...
+            - repo_name: credentials-themes
+            ...
+
+5. Create a pull request in the `openedx-translations repo`_ and wait for review and approval.
+   Once the pull request is merged the GitHub/Transifex workflow will be ongoing as described in the
+   Introduction section. No additional configuration is needed.
+
+6. Verify the workflow is syncing the translations properly as described in the Debugging section below.
+
+7. Install the XBlock or plugin in a devstack and run the ``OPENEDX_ATLAS_PULL=true make pull_translations`` in
+   the edx-platform repo to preview the translations in devstack.
+
+   This step is somewhat oversimplified and assumes you're familiar with the Open edX development workflow whether
+   you're using `Tutor <https://docs.tutor.overhang.io/>`_ or the
+   `Open edX devstack <https://github.com/openedx/devstack/>`_.
```

Comment:
does it make sense to move this step above the PR step? it seems like testing locally is something that should happen before PRing, but the listed command won't work pre-PR